src/Sulu/Component/CustomUrl/Routing/RouteProvider.php

```diff
@@ -0,0 +1,194 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\CustomUrl\Routing;
+
+use PHPCR\Util\PathHelper;
+use Sulu\Bundle\ContentBundle\Document\RouteDocument;
+use Sulu\Component\Content\Document\WorkflowStage;
+use Sulu\Component\CustomUrl\Manager\CustomUrlManagerInterface;
+use Sulu\Component\DocumentManager\PathBuilder;
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Symfony\Cmf\Component\Routing\RouteProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * Provides custom-url routes.
+ */
+class RouteProvider implements RouteProviderInterface
```

Comment:
Maybe call that one `CustomUrlRouteProvider` so that it matches the style of the `ContentRouteProvider`, and to make the name more unique?